### PR TITLE
Add Biome decorator

### PR DESCRIPTION
**Pull request by jimmikaelkael**
This is extensible and could help to implement the other things to decorate biomes with variations.
It already contains:
- tall grass (and ferns but only in some biomes, so will be visible later...)
- flowers and double flowers
- sugar canes
- dead bushes
- pumpkins
- melons
- water lilies
- cactuses

grass and flowers

---

_Edited by turt2live_

**Related Links:**
- Ticket: #76

### CODE REVIEW

**Comment by Tonodus** [comment]
Some thoughts:
- BlockDecorator could extend BlockPopulator, renaming #decorate() to #populate().
- PumpkinDecorator doesn't use functions provided by Decorator, so it could extend BlockPopulator.
- the `for (int i = 0;i < getAmount(...);i++)` could probably be moved into the Decorator class, so that the decorator overrides populate and calls decorate that specific amount times.

**Comment by jimmikaelkael** [comment]
Thank you ;)
I'll try to make some improvements tomorrow.

**Comment by jimmikaelkael** [comment]
@Tonodus made changes with your suggestions, they were very appropriate, thank you.

**Comment by jimmikaelkael** [comment]
I stop to add things and let this ready for review. 

I have vanilla-like dungeons with their chests and content ready too but I think it's more a populator than a decorator, so I'll put it in another PR.

EDIT: just one note, I've let a debug line when a desert well is generated so that we can easily locate it for the tests.

**Comment by jimmikaelkael** [comment]
Please note that this PR makes net.glowstone.generator.populators.DesertPopulator and net.glowstone.generator.populators.FlowerPopulator obsolete, but I prefer to keep this PR more standalone.

File: src/main/java/net/glowstone/generator/objects/DesertWell.java [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        world.getBlockAt(sourceX - 1, sourceY, sourceZ).setType(Material.STATIONARY_WATER);
+        world.getBlockAt(sourceX - 1, sourceY, sourceZ).setData((byte) 0);
+        world.getBlockAt(sourceX + 1, sourceY, sourceZ).setType(Material.STATIONARY_WATER);
+        world.getBlockAt(sourceX + 1, sourceY, sourceZ).setData((byte) 0);
+        world.getBlockAt(sourceX, sourceY, sourceZ - 1).setType(Material.STATIONARY_WATER);
+        world.getBlockAt(sourceX, sourceY, sourceZ - 1).setData((byte) 0);
+        world.getBlockAt(sourceX, sourceY, sourceZ + 1).setType(Material.STATIONARY_WATER);
+        world.getBlockAt(sourceX, sourceY, sourceZ + 1).setData((byte) 0);
+
+        GlowServer.logger.info("desert well generated: " + sourceX + "," + sourceY + "," + sourceZ);
```
Comment (turt2live) [inline src/main/java/net/glowstone/generator/objects/DesertWell.java]: `*cough*`

Comment (jimmikaelkael) [inline src/main/java/net/glowstone/generator/objects/DesertWell.java]: That was intentional for the tests so you can locate it more easily ^^.
I'll remove it.

**Comment by SpaceManiac** [comment]
Merged in 43b30d91689a67ff41b1dd227b4bc5f59f2d8697.

**Pull request closed without merging**